diff --git a/scalars/formats/index.js b/scalars/formats/index.js
index 195d9a1cbd7561e6394e890056d3a1b7eedddcca..639dcf048266b4cb59c5496d295a23d900e68f27 100644
--- a/scalars/formats/index.js
+++ b/scalars/formats/index.js
@@ -1,17 +1,21 @@
-const formats = [
-  'byte',
-  'date-time',
-  'date',
-  'email',
-  'ipv4',
-  'ipv6',
-  'uri',
-  'uuid'
-]
-const fns = {}
+const byte = require('./byte');
+const time = require('./date-time');
+const date = require('./date');
+const email = require('./email');
+const ipv4 = require('./ipv4');
+const ipv6 = require('./ipv6');
+const uri = require('./uri');
+const uuid = require('./uuid');

-formats.forEach(function (format) {
-  fns[format] = require('./' + format)
-})
+ fns = {
+  byte,
+  'date-time': time,
+  date,
+  email,
+  ipv4,
+  ipv6,
+  uri,
+  uuid
+}

 module.exports = fns
